### ISSUE TITLE
C and Perl are not sexy.

### ISSUE THREAD

**Issue by fuzzynoise** [zmanda/amanda#118]
I have issue with too few issues.

I believe from the core of my being that backup is important. I KNOW that tape backup is cheap while effective and I know that Amanda offers a flexibility that is ABSOLUTELY invaluable. I want Amanda to progress into the 21st century tho, and a lot of that requires EG translations to Python over Perl and...smart bois looking at C so we can write either better C or ASM. Obviously, I don't think machine problems are solved by floating away from the machine. We need some bare-metal. I know there is a commercial UI....but what if it was a user UI? I only ask since I forget what the U in UI stands for...

What do people think? What would you want? What is your use case?

**Comment by chassell** [comment]
MOARR BUZZWORDS!   WRITE IT ALL IN NODE.JS AND THEN OPTIMIZE IT IN THE CLOUD!!   PAY AMAZON EVERY SINGLE DOLLAR YOU ...  oh wait not supposed to write that down... oops..

**Comment by fuzzynoise** [comment]
That's a lovely idea. I don't think that the next best thing is the best
thing...but I think the old thing is old.

**Comment by fuzzynoise** [comment]
I would appreciate a constructive discourse. Let's figure it out. I have written parts into the commercial parts of this thing. I like that the currently commercial aspect is opposed to it. What would users enjoy? This is a user discussion.

**Comment by chassell** [comment]
For my part .... the issues of changing language base come down to this:

1) Avoid tech-debt tangles that keep bugs/misfeatures from being ever fixed or extended.  (REAL!!)

2) Avoid legacy problems with many modules being unchangeable due to interconnection to others

3) Allow dynamic UI events but retain SELinux/Linux/OS-Insensitive CLI systems

4) Yes, allowing Python / Go  and (cringe) JScript event APIs in someday may be a futureproof key

Genuinely and truly... there's a reason many other languages (and NPM!!) are *not* wise in the infrastructure-and-OS-domain.   No matter how many docker versions or VMs one creates ... you are *only* intending to securely snarf up and then retrieve [maybe] one little part of a backup.   That's big raw data and simple, crucial, no-dicey-systems permissions allowed.

Modern architectures are, well, good for making data-sniffing UI sites.... because money.   M$ already tried to make C# (and others tried Java) as their systems language and it failed HARD.   It's hardcore languages or nothing for the ins and outs of raw bytes.

**Comment by chassell** [comment]
TL;DR:  C and PERL ... are not sexy.  Neither is fishing a palette out of a warehouse.  :-].

**Comment by fuzzynoise** [comment]
Obviously, we're still looking relatively low RE data moving since...data is a low level abstraction. You want to move ones and zeros? Let's move 'em. 

> Avoid tech-debt tangles that keep bugs/misfeatures from being ever fixed or extended. (REAL!!)

Yes, this is wildly correct. There are bugs up and down. How do we write it so folks between 15 and 40 understand where the bugs are?

> Allow dynamic UI events and failure detection but retain SELinux/Linux/OS-Insensitive CLI API

Absolutely. This can be automated to a large extent with consent.

> Yes, allowing Python / Go and (cringe) JScript event APIs in someday may be a futureproof key

Yes. There should be easily built-in plugins. I don't know what language should be the thing...but something currently. I don't think you can future-proof without idiot proofing and those folks are ingenious.
